packages/claroty_ctd/data_stream/asset/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,1338 @@
+            return Long.parseLong(value);
+          } else if (value instanceof Number) {
+            return ((long) value).longValue();
+          } else {
+            throw new Exception('Unsupported type');
+          }
+        }
+        def convertToBoolean(def value) {
+          if (value instanceof String) {
+            return Boolean.parseBoolean(value);
+          } else if (value instanceof Boolean) {
+            return (Boolean) value;
+          } else {
+            throw new Exception('Unsupported type');
+          }
+        }
+        def renameKeys(Map json, Map keyMap, List longFields, List stringFields, List boolFields) {
+          def updatedJson = new HashMap();
+          for (def entry: json.entrySet()) {
+            def key = entry.getKey();
+            def value = entry.getValue();
+            if (value instanceof Map) {
+              if (keyMap.containsKey(key)) {
+                updatedJson[keyMap[key]] = renameKeys(value, keyMap, longFields, stringFields, boolFields);
+              } else {
+                updatedJson[key] = renameKeys(value, keyMap, longFields, stringFields, boolFields);
+              }
+            } else if (value instanceof List) {
+              def updatedList = [];
+              for (def item: value) {
+                if (item instanceof Map) {
+                  updatedList.add(renameKeys(item, keyMap, longFields, stringFields, boolFields));
+                } else {
+                  updatedList.add(item);
+                }
+              }
+              if (keyMap.containsKey(key)) {
+                updatedJson[keyMap[key]] = updatedList;
+              } else {
+                updatedJson[key] = updatedList;
+              }
+            } else {
+              if (keyMap.containsKey(key)) {
+                if (longFields.contains(keyMap[key])) {
+                  updatedJson[keyMap[key]] = convertToLong(value);
+                } else if (stringFields.contains(keyMap[key]) && value != null) {
+                  updatedJson[keyMap[key]] = value.toString();
+                } else if (boolFields.contains(keyMap[key])) {
+                  updatedJson[keyMap[key]] = convertToBoolean(value);
+                } else {
+                  updatedJson[keyMap[key]] = value;
+                }
+              } else {
+                updatedJson[key] = value;
+              }
+            }
+          }
+          return updatedJson;
+        }
+        ctx.claroty_ctd.asset.put('children', new ArrayList());
+        def longFields = new ArrayList(params.longfield);
+        def stringFields = new ArrayList(params.stringfield);
+        def boolFields = new ArrayList(params.boolfield);
+        def renamefields = new HashMap(params.renamefield);
+        for (child in ctx.json.children) {
+          def children = renameKeys(child, renamefields, longFields, stringFields, boolFields);
+          ctx.claroty_ctd.asset.children.add(children);
+        }
```

I don't think we need these, just pass the `param. …` value.

```suggestion
        ctx.claroty_ctd.asset.put('children', new ArrayList());
        for (child in ctx.json.children) {
          def children = renameKeys(child, params.renamefield, params.longfield, params.stringfield, params.boolfield);
          ctx.claroty_ctd.asset.children.add(children);
        }
```